raami/queries.js

```diff
@@ -0,0 +1,269 @@
+// };
+
+// var pgp = require("pg-promise")(options);
+// var db = pgp(process.env.DATABASE_URL)
+
+
+module.exports = {
+  getPeople,
+	getArtists,
+	createArtist,
+	getArtist,
+	updateArtist,
+	getWork,
+	getWorks,
+	createWork,
+	updateWork,
+	removeWork,
+};
+
+function getPeople(req, res, next) {
+  var _id = (req.params.id);                             
+  req.app.locals.db.any(`
+    SELECT id, person_id, name, continent, url, filename, filedata, description,
+           transport, legal, auction, print, digital, agent, contact, waitlist, postage
+      FROM Artist
+     WHERE person_id = $1`, _id
+  )
+    .then(function (data) {
+      res.status(200)
+        .json(data);
+    })
+    .catch(function (err) {
+      return next(err);
+    });
+}
+
+function getArtists(req, res, next) {
+  req.app.locals.db.any('SELECT person_id, id, name FROM Artist')
+    .then(function (data) {
+      res.status(200)
+        .json(data);
+    })
+    .catch(function (err) {
+      return next(err);
+    });
+}
+
+function getArtist(req, res, next) {
+  var _id = (req.params.id);
+  req.app.locals.db.one(`
+    SELECT *
+      FROM Artist
+     WHERE id = $1`, _id
+  )
+    .then(function (data) {
+      res.status(200)
+        .json(data);
+    })
+    .catch(function (err) {
+      return next(err);
+    });
+}
+
+
+function createArtist(req, res, next) {
+
+  req.app.locals.db.one(`
+     INSERT INTO Artist
+                 (
+                   person_id, name, continent, url, filename, filedata,
+                   description, transport, legal, auction, print, digital, agent, contact, waitlist, postage
+                 )
+          VALUES (
+                   $(person_id), $(name), $(continent), $(url), $(filename),
+                   $(filedata), $(description), $(transport), $(legal),
+                   ($(auction)), ($(print)), $(digital), $(agent), $(contact), $(waitlist), ($(postage))
+                 )
+       RETURNING id`, req.body
+  )
+    .then(function (data) {
+      console.log(data.id)
+      res.status(200)
+        .json({
+          status: 'success',
+          inserted: data.id
+        });
+    })
+    .catch(function (err) {
+      return next(err);
+    });
+}
+
+
+function updateArtist(req, res, next) {
+  var _id = (req.params.id);
+  req.app.locals.db.none(`
+    UPDATE Artist
+       SET continent=$1, url=$2, filename=$3, filedata=$4, name=$5,
+           description=$6, transport=$7, legal=$8, auction=$9, print=$10,
+           digital=$11, agent=$12, contact=$13, waitlist=$14, postage=$15
+     WHERE id=$16`, [
+      req.body.continent,
+      req.body.url,
+      req.body.filename,
+      req.body.filedata,
+      req.body.name,
+      req.body.description,
+      req.body.transport,
+      req.body.legal,
+      (req.body.auction),
+      (req.body.print),
+      req.body.digital,
+      req.body.agent,
+      req.body.contact,
+      req.body.waitlist,
+      (req.body.postage),
+      _id
+    ]
```

Comment:
Again, extra parentheses.

Should require `raami_admin` rights, or check that the email address of `person_id` matches `req.session.user.email`.

Also, would be clearer with an object rather than a 16-value array. Take a look at the pg-promise helpers [`sets`](http://vitaly-t.github.io/pg-promise/helpers.html#.sets) and [`update`](http://vitaly-t.github.io/pg-promise/helpers.html#.update).